src/main.cpp

```diff
@@ -238,11 +233,70 @@ try
 
     config->set_config_filename("unity-system-compositor.conf");
 
-    config->apply_settings();
+    // settings for the display manager connection
+    int from_dm_fd;
+    int to_dm_fd;
+    std::string debug_active_session_name;
+
+    auto from_dm_fd_flag = pre_init(miral::CommandLineOption(
+        [&](int from_dm_file_descriptor)
+        {
+            from_dm_fd = from_dm_file_descriptor;
+        },
+        "from-dm-fd",
+        "File descriptor of read end of pipe from display manager [int]",
+        -1));
+    from_dm_fd_flag(*config);
+
+    auto to_dm_fd_flag = pre_init(miral::CommandLineOption(
+        [&](int to_dm_file_descriptor)
+        {
+            to_dm_fd = to_dm_file_descriptor;
+        },
+        "to-dm-fd",
+        "File descriptor of write end of pipe to display manager [int]"
+        -1));
+    to_dm_fd_flag(*config);
 
-    auto the_session_switcher = [&session_switcher]() { return session_switcher; };
+    auto display_stub_flag = pre_init(miral::CommandLineOption(
+        [&](std::string const& display_stub_name)
+        {
+            debug_active_session_name = display_stub_name;
+        },
+        "debug-active-session-name",
+        "Expected connection when run without a display manager (only useful when debugging)",
+        "nested-mir@:/run/user/1000/mir_socket"));
+    display_stub_flag(*config);
+
+    // the display manager connection
+    std::shared_ptr<usc::DMConnection> dm_connection;
+
+    miral::CommandLineOption debug_without_dm_flag{
+        [&](bool debug_without_dm)
```

Comment:
At a guess, this is only being invoked if the flag is actually set -- which has always been the case in my local tests, but stopped being true once I took it out for a spin on my Unity8 system.  So it might be a good idea to only set a flag using the `CommandLineOption`, and use an explicit `AddInitCallback` to initialize `dm_connection`.

Reply:
The remark here seems to be completely off base.  Current version of code works without needing to change this particular `CommandLineOption`.